### DISCUSSION TITLE
Issue Running Next.js Behind Proxies

### DISCUSSION THREAD

**Discussion by kjschulz** [vercel/next.js#22548] [Help]
My prod app server runs apps in Kubernetes behind an nginx proxy, orchestrated by Dokku.  So far I've always used express.js and simply informed express that the `X-Forwarded-*` header fields may be trusted by using `app.set('trust proxy', true)`.

This is my first time using Next.js for an app in production.  Is there a `trust proxy` equivalent purely in Next.js?  I would really prefer not to have to wrap the Next.js app in an express server just so that I can call `app.set('trust proxy', true)`.

What's the best practice for this?

**Comment by gijsbotje** [comment]
@kjschulz We are having the same issue with this header. We want to enable domain routing but use the same container for it. Unfortunately, I haven't found a way to make it possible to use the `x-forwarded-host` header. It only looks at the host or hostname header.
@danielyefet has tried to add the exact option you mentioned in https://github.com/vercel/next.js/pull/20662. But the PR was closed because there was no feature request in the issues. I created an issue a few days ago, but it was transformed to a discussion (https://github.com/vercel/next.js/discussions/34571). 

It's a pretty easy fix i'd say, but I guess the team thinks differently about it.

**Reply by kjschulz** [reply]
Thanks @gijsbotje, with the lack of resolution and other performance issues we experienced while developing on the framework, we moved on and are no longer using Next.js
